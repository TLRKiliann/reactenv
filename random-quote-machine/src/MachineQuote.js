import React from 'react';
import $ from 'jquery';
import { FaTwitter } from "react-icons/fa";
import './styles.scss';
import './App.css';


class MachineQuote extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      textQuote: '',
      authorQuote: '',
      colorText: ['lightpink', 'cyan', 'yellow'],
      colorQuote: ["grey", "blue", "orange", "green", "red", "purple"],
      colorsBody: ["lightgrey", "lightblue", "lightsalmon", "lightgreen",
        "lightcoral", "lightpink"],
      isLoaded: false
    };
    this.jQuerycode = this.jQuerycode.bind(this);
    //this.customNewQuote.bind(this) => already run with button
  }

  componentDidMount() {
    const catchData = this;
    fetch('https://type.fit/api/quotes')
      .then(response => {
        return response.json();
      })
      .then(function(data) {
        const index = Math.floor((Math.random() * data.length) + 1);
        catchData.setState({
          isLoaded: true,
          textQuote: data[index].text,
          authorQuote: data[index].author
        });
      });
  }

  jQuerycode = () => {
    const { colorText, colorQuote, colorsBody } = this.state;
    const spectralColor = (colorQuote[Math.floor((Math.random() * 6))]);
    const changeBodyCol = (colorsBody[Math.floor((Math.random() * 6))]);
    const spectralText = (colorText[Math.floor((Math.random() * 3))]);
    console.log('2colorsBody : ', spectralColor, spectralColor, spectralText)
    $("button").ready(function () {
      $("body").css({ "background-color": changeBodyCol});
      $(".turn--class").css({ "background-color": spectralColor});
      $(".turn--class").css({"color": spectralText});
    })
    this.componentDidMount();
  };

  customNewQuote() {
    this.jQuerycode()
  }

  render() {
    const { isLoaded, textQuote, authorQuote } = this.state;
    if (!isLoaded) {
      return <h2>No data loaded !</h2>
    } else {
      return (
        <div className="machine--class">

          <div className='turn--class'>
            <div id='quote-box'>
              <div id='text'>{textQuote}</div>
              <div id='author'>{authorQuote}</div>
            </div>

            <div id='new-quote'>
              <button 
                onClick={this.customNewQuote.bind(this)}
                className='btn btn--custom'>New Quote</button>

              <a className="btn btn-primary"
                 data-bs-toggle="collapse"
                 href="twitter.com/intent/tweet"
                 target="_blank"
                 rel="noreferrer"
                 role="button"
                 aria-expanded="false"
                 aria-controls="collapseExample">
                <FaTwitter />
              </a>
            </div>
          </div>
        </div>
      );
    }
  }
}

export default MachineQuote;

//href="https://twitter.com/intent/tweet/?hashtag={ this.state.textQuote }"

//onClick={this.customNewQuote}