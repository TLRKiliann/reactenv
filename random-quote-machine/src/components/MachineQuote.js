import React from 'react';
import $ from 'jquery'
import { FaTwitter } from "react-icons/fa";


class MachineQuote extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      textQuote: '',
      authorQuote:'',
    };
    this.customNewQuote = this.customNewQuote.bind(this);
    this.handleNewQuote = this.handleNewQuote.bind(this)
  }

  jQuerycode = () => {
    const colors = ["grey", "blue", "orange", "green", "red", "purple"];
    const colorsBody = ["grey", "lightblue", "lightorange", "lightgreen", "lightred", "lightpink"];
    const colorText = ['purple', 'white', 'black'];

    const spectralcolor = (colors[Math.floor((Math.random() * 6) + 1)]);
    const bodyColor = (colorsBody[Math.floor((Math.random() * 5) + 1)]);
    const spectralText = (colorText[Math.floor((Math.random() * 3) + 1)]);

    $("button").click(function () {
      $("#quote-box").css({ backgroundColor : spectralcolor})
      $("#quote-box").css({color: spectralText})
      $("body").css({ backgroundColor : bodyColor})

    })
    this.handleNewQuote()
  };

  customNewQuote() {
    this.jQuerycode()
  }

  handleNewQuote() {
    const app = this;
    fetch('https://type.fit/api/quotes')
      .then(response => {
        return response.json();
      })
      .then(function(data) {
        const index = Math.floor((Math.random() * data.length) + 1);
        app.setState({
          textQuote: data[index].text,
          authorQuote: data[index].author
        });
      });
  }
  render() {
    return (
      <div className="machine--class">
        <h1>Random Quote Machine</h1>

        <div id='quote-box'>
          <div id='text'>{this.state.textQuote}</div>
          <div id='author'>{this.state.authorQuote}</div>
        </div>

        <div id='new-quote'>
          <button onClick={this.customNewQuote}>New Quote</button>

          <a className="btn btn-primary"
             data-bs-toggle="collapse"
             href="https://twitter.com/intent/tweet/?text={ this.state.textQuote }"
             target="_blank"
             rel='noreferrer'
             role="button"
             aria-expanded="false"
             aria-controls="collapseExample">
            <FaTwitter />
          </a>
        </div>
      </div>
    );
  }
}

export default MachineQuote;
