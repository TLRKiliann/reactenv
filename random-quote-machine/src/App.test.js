import React from 'react'
import {render, cleanup} from '@testing-library/react'
import MachineQuote from './MachineQuote'

 afterEach(cleanup)
 
 it('should take a snapshot', () => {
    const { asFragment } = render(<MachineQuote />)
    
    expect(asFragment(<MachineQuote />)).toMatchSnapshot()
});
