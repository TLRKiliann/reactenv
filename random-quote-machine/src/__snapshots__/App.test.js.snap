// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <h2>
    No data loaded !
  </h2>
</DocumentFragment>
`;
